### PR TITLE
Better document VectorMemory::Pointer.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

File: include/deal.II/lac/vector_memory.h [hunk 1]
```diff
@@ -54,7 +55,8 @@ DEAL_II_NAMESPACE_OPEN
  * vectors to using classes.
  *
  * For example, the PrimitiveVectorMemory class simply allocates and
- * deallocated vectors each time it is asked for a vector. It is an
+ * deallocated vectors via the operating system facilities (i.e., using
```
Comment (kronbichler) [inline include/deal.II/lac/vector_memory.h]: deallocated -> deallocates

File: include/deal.II/lac/vector_memory.h [hunk 2]
```diff
@@ -63,10 +65,38 @@ DEAL_II_NAMESPACE_OPEN
...
+ * <h3> Practical use </h3>
+ *
+ * Classes derived from this base class return pointers to new vectors
+ * via the VectorMemory::alloc() function, and re-claim the vector
+ * when it is returned via VectorMemory::free(). These two functions
+ * therefore play a similar role as @p new and @p delete. This includes
+ * the usual drawbacks: It is simple to forget to call VectorMemory::free()
+ * at the end of a function that uses this facility, or to forget it in an
+ * @p if branch of the function where one has an early @p return from
+ * the function. In both cases, a memory leak results: a correct piece of
```
Comment (kronbichler) [inline include/deal.II/lac/vector_memory.h]: I'm not a native speaker, but "a memory leak results" sounds weird to me. Wouldn't it be "a memory leak occurs"?

Comment (jppelteret) [reply include/deal.II/lac/vector_memory.h]: I think that "occurs" sounds a _bit_ more natural in this context. When using "results" I would likely have written it as "In both cases, this results / would result in a memory leak"

Comment (kronbichler) [reply include/deal.II/lac/vector_memory.h]: The latter two would also be the choices I would have made with "results", thanks for the confirmation @jppelteret ;-)

**Comment by bangerth** [comment]
I've updated the text as suggested -- thanks for the careful reading and the suggestions!

**Pull request merged**